notify/notify_test.go

```diff
@@ -283,6 +283,49 @@ func TestRoutingStage(t *testing.T) {
 	}
 }
 
+type notifierConfigFunc func() bool
+
+func (f notifierConfigFunc) SendResolved() bool {
+	return f()
+}
+
+type RecordNotifier struct {
+	alerts []*types.Alert
+}
```

Comment:
I've found over time that having a single test/mock implementation of an interface is easier here. Usually, at some point there'll be further tests requiring a mock as well wis slightly different behavior.
It's then easier to have:

```go
type mockNotifier struct {
    notify func(context.Context, alerts ...*types.Alert) (bool, error)
}
```

And in your actual test you instantiate:

```go
var recorded []*types.Alert
n := &mockNotifier{
    notify: func(_ context.Context, alerts ...*types.Alert) (bool, error) {
        recorded = append(recorded, alerts...)
    },
}
// test code ...
```

It's more verbose but it more flexible and makes the entire behavior easily visible within the test function.